pageserver/src/tenant/secondary/heatmap.rs

```diff
@@ -0,0 +1,64 @@
+use std::time::SystemTime;
+
+use crate::tenant::{
+    remote_timeline_client::index::IndexLayerMetadata, storage_layer::LayerFileName,
+};
+
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr, TimestampSeconds};
+
+use utils::{generation::Generation, id::TimelineId};
+
+#[derive(Serialize, Deserialize)]
+pub(super) struct HeatMapTenant {
```

Comment:
My bias here is to avoid treading on the names of types we use extensively throughout the code: not every type name has to be unique, but `Tenant` is such a big one that I want to avoid having multiple types with that name (in #6123 I add `SecondaryTenant` rather than `secondary::Tenant` for the same reason)